zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineCache.kt

```diff
@@ -0,0 +1,144 @@
+package app.cash.zipline.loader
+
+import kotlinx.datetime.Clock
+import okio.ByteString
+import okio.FileSystem
+import okio.Path
+import okio.Path.Companion.toPath
+
+/**
+ * Zipline Cache State Machine
+ *
+ *            download
+ *              start
+ *         --------------->
+ *   ABSENT                DIRTY
+ *         <---------------
+ *    ^       download       | download
+ *    |        failed        | success
+ *    |                      |
+ *    |                      v
+ *    '------------------ READY
+ *           pruned
+ *
```

Comment:
```suggestion
 * ```
 *            download
 *              start
 *         --------------->
 *   ABSENT                DIRTY
 *         <---------------
 *    ^       download       | download
 *    |        failed        | success
 *    |                      |
 *    |                      v
 *    '------------------ READY
 *           pruned
 * ```
```